src/logical/builder_proxy.rs

```diff
@@ -13,12 +13,16 @@ use crate::{
 
 /// Trait capturing builder proxies that use plain string (used for parsing in logical layer)
 pub trait LogicalColumnBuilderProxy<'a, 'b>: ColumnBuilderProxy<String> {
-    /// create a new LogicalColumnBuilerProxy from the physical equivalent
+    /// create a new [`LogicalColumnBuilderProxy`] which nests its physical target equivalent [`BuilderProxy`][crate::physical::builder_proxy::PhysicalBuilderProxyEnum].
+    ///
+    /// # Panics
+    /// If the logical and the nested physical type are not compatible.
     fn new(physical_builder_proxy: &'b mut PhysicalBuilderProxyEnum<'a>) -> Self
     where
         Self: Sized;
 }
 
+/// implement the [`ColumnBuilderProxy`] trait methods [`commit`][ColumnBuilderProxy::commit] and [`forget`][ColumnBuilderProxy::forget]
```

Comment:
Probably better not to name the specific methods. What's important is that this implements the type-independent methods of the trait.